pep-0539.txt

```diff
@@ -24,44 +24,106 @@ included in ``Python.h`` either directly or indirectly), this proposal probably
+    Storage and should not be confused with "Transportation Layer Security"
+    protocols.
+
 
 Specification
 =============
 
-The current API for TLS used inside the CPython interpreter consists of 5
+The current API for TLS used inside the CPython interpreter consists of 6
 functions::
 
     PyAPI_FUNC(int) PyThread_create_key(void)
     PyAPI_FUNC(void) PyThread_delete_key(int key)
     PyAPI_FUNC(int) PyThread_set_key_value(int key, void *value)
     PyAPI_FUNC(void *) PyThread_get_key_value(int key)
     PyAPI_FUNC(void) PyThread_delete_key_value(int key)
+    PyAPI_FUNC(void) PyThread_ReInitTLS(void)
 
 These would be superseded by a new set of analogous functions::
 
     PyAPI_FUNC(int) PyThread_tss_create(Py_tss_t *key)
-    PyAPI_FUNC(void) PyThread_tss_delete(Py_tss_t key)
+    PyAPI_FUNC(void) PyThread_tss_delete(Py_tss_t *key)
     PyAPI_FUNC(int) PyThread_tss_set(Py_tss_t key, void *value)
     PyAPI_FUNC(void *) PyThread_tss_get(Py_tss_t key)
     PyAPI_FUNC(void) PyThread_tss_delete_value(Py_tss_t key)
+    PyAPI_FUNC(void) PyThread_ReInitTSS(void)
+
+The specification also adds three new features:
+
+* A new type ``Py_tss_t``--an opaque type the definition of which may
+  depend on the underlying TLS implementation.  It is defined::
+
+      typedef struct {
+          bool _is_initialized;
+          NATIVE_TLS_KEY_T _key;
+      } Py_tss_t;
+
+  where ``NATIVE_TLS_KEY_T`` is a macro whose value depends on the
+  underlying native TLS implementation (e.g. ``pthread_key_t``).
+
+* A constant default value for ``Py_tss_t`` variables,
+  ``Py_tss_NEEDS_INIT``.
+
+* A new inline function::
+
+      static inline bool PyThread_tss_is_created(Py_tss_t key)
 
-along with a new type ``Py_tss_t``--an opaque type the definition of which
-is undefined here, and may depend on the underlying TLS implementation.
+  which returns ``true`` if the given ``Py_tss_t`` has been initialized
+  (i.e. by ``PyThread_tss_create``).
 
 The new ``PyThread_tss_`` functions are almost exactly analogous to their
 original counterparts with a minor difference:  Whereas
 ``PyThread_create_key`` takes no arguments and returns a TLS key as an
 ``int``, ``PyThread_tss_create`` takes a ``Py_tss_t*`` as an argument, and
-returns a ``Py_tss_t`` by pointer--the ``int`` return value is a status,
+returns a ``Py_tss_t`` by pointer.  The behavior of ``PyThread_tss_create``
+is undefined if the value pointed to by the ``key`` argument is not
+initialized by ``Py_tss_NEEDS_INIT``.  The ``int`` return value is a status,
 returning zero on success and non-zero on failure.  The meanings of non-zero
-status codes are not not defined by this specification.
+status codes are not defined by this specification.
+
+Similarly ``PyThread_tss_delete`` is passed a ``Py_tss_t*`` whereas
+previouly the key was passed to ``PyThread_delete_key`` by value.
 
 The old ``PyThread_*_key*`` functions will be marked as deprecated.
```

Comment:
Same.  Let me double-check on what the implementation has (I'm pretty sure also just a documentation deprecation).